src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -439,6 +439,40 @@ public async Task<IActionResult> GetCompatibility(string sessionId)
             return Ok(output);
         }
 
+        /// <summary>
+        /// Start a dry run of the deployment, which performs all the steps in the process of performing a deployment except for running 'cdk bootstrap' and 'cdk deploy'.
+        /// This includes, but is not limited to, creating a deployment bundle and generating a CDK project.
+        /// </summary>
+        [HttpPost("session/<sessionId>/dryrun")]
+        [SwaggerOperation(OperationId = "StartDryRunDeployment")]
+        [SwaggerResponse(200, type: typeof(StartDryRunDeploymentOutput))]
+        [Authorize]
+        public async Task<IActionResult> StartDryRunDeployment(string sessionId)
+        {
+            var state = _stateServer.Get(sessionId);
+            if (state == null)
+            {
+                return NotFound($"Session ID {sessionId} not found.");
+            }
+
+            var serviceProvider = CreateSessionServiceProvider(state);
+
+            var orchestratorSession = CreateOrchestratorSession(state);
+
+            var orchestrator = CreateOrchestrator(state, serviceProvider);
+
+            var cdkProjectHandler = CreateCdkProjectHandler(state, serviceProvider);
+
+            if (state.SelectedRecommendation == null)
```

Comment:
There should be another error if the selected recommendation is not CDK based.

Reply:
done